syntax = "proto3";
package deploy;




import "easy_flow_sdk/model/easy_flow/deploy_label.proto";
import "easy_flow_sdk/model/easy_flow/deploy_target.proto";
import "easy_flow_sdk/model/cmdb/cluster_info.proto";
import "easy_flow_sdk/model/easy_flow/version_info.proto";
import "easy_flow_sdk/model/easy_flow/target_info.proto";
import "easy_flow_sdk/model/easy_flow/package_info.proto";

/*
Create请求
*/
message CreateRequest {
message Batches {
    /*
    批次主机列表
    */
    repeated easy_flow.DeployTarget targets = 1 ;
}
        
message InstanceInfo {
    /*
    版本名
    */
    string versionName = 1 ;
    /*
    版本信息
    */
    easy_flow.VersionInfo versionInfo = 2 ;
    /*
    包Id
    */
    string packageId = 3 ;
    /*
    安装路径
    */
    string installPath = 4 ;
    /*
    版本Id
    */
    string versionId = 5 ;
}
        
message OperationInfo {
    /*
    操作信息
    */
    string operation = 1 ;
    /*
    当前版本信息
    */
    easy_flow.VersionInfo versionToInfo = 2 ;
    /*
    升级版本信息
    */
    easy_flow.VersionInfo versionFromInfo = 3 ;
    /*
    安装路径
    */
    string installPath = 4 ;
    /*
    包Id
    */
    string packageId = 5 ;
}
        
message ConfigList {
message Configs {
message Items {
    /*
    名称
    */
    string name = 1 ;
    /*
    路径
    */
    string path = 2 ;
}
        
    
    /*
    包Id
    */
    string packageId = 1 ;
    /*
    选项列表
    */
    repeated Items items = 2 ;
    /*
    安装路径
    */
    string installPath = 3 ;
}
        
    
    /*
    主机列表
    */
    repeated string hosts = 1 ;
    /*
    配置信息
    */
    repeated Configs configs = 2 ;
}
        
message ConfigDiff {
message Detail {
message Items {
    /*
    路径
    */
    string path = 1 ;
    /*
    包新名称
    */
    string newName = 2 ;
    /*
    包原名称
    */
    string oldName = 3 ;
}
        
    
    /*
    详情
    */
    repeated Items items = 1 ;
    /*
    包Id
    */
    string packageId = 2 ;
    /*
    安装路径
    */
    string installPath = 3 ;
}
        
    
    /*
    主机列表
    */
    repeated string hosts = 1 ;
    /*
    包配置变更信息
    */
    repeated Detail detail = 2 ;
}
        
    
    /*
    是否需要通知
    */
    bool needNotify = 1 ;
    /*
    标签
    */
    easy_flow.DeployLabel labels = 2 ;
    /*
    应用Id
    */
    string appId = 3 ;
    /*
    应用名称
    */
    string appName = 4 ;
    /*
    集群Id
    */
    string clusterId = 5 ;
    /*
    集群类型
    */
    string clusterType = 6 ;
    /*
    自动分批:每批次部署机器数;手动分批:部署总共批次
    */
    int32 batchNum = 7 ;
    /*
    部署间隔时长(s)
    */
    int32 batchInterval = 8 ;
    /*
    分批详情
    */
    repeated Batches batches = 9 ;
    /*
    是否失败暂停
    */
    bool failedStop = 10 ;
    /*
    目标设备Id
    */
    string targetId = 11 ;
    /*
    目标设备名称
    */
    string targetName = 12 ;
    /*
    实例Id
    */
    string instanceId = 13 ;
    /*
    集群列表
    */
    cmdb.ClusterInfo cluster = 14 ;
    /*
    目标设备上次成功部署的实例信息
    */
    repeated InstanceInfo instanceInfo = 15 ;
    /*
    操作信息
    */
    repeated OperationInfo operationInfo = 16 ;
    /*
    部署主机信息列表
    */
    repeated easy_flow.TargetInfo targetList = 17 ;
    /*
    部署包信息列表
    */
    repeated easy_flow.PackageInfo packageList = 18 ;
    /*
    配置列表
    */
    repeated ConfigList configList = 19 ;
    /*
    配置包Id
    */
    string configPackageId = 20 ;
    /*
    配置变更信息
    */
    repeated ConfigDiff configDiff = 21 ;
}

/*
Create返回
*/
message CreateResponse {
message Data {
    /*
    任务Id
    */
    string taskId = 1 ;
}
        
    
    /*
    错误码
    */
    int32 code = 1 ;
    /*
    消息
    */
    string msg = 2 ;
    /*
    返回数据
    */
    Data data = 3 ;
}

/*
CreateApi返回
*/
message CreateResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    deploy.CreateResponse data = 4 ;
}

